content/library/api/testing/index.md

```diff
@@ -0,0 +1,330 @@
+---
+title: App testing
+slug: /library/api-reference/app-testing
+---
+
+# App testing
+
+Streamlit app testing framework enables developers to build and run headless tests that execute their app code directly, simulate user input, and inspect rendered outputs for correctness.
+
+The provided class, AppTest, simulates a running app and provides methods to set up, manipulate, and inspect the app contents via API instead of a browser UI. It can be used to write automated tests of an app in various scenarios. These can then be run using a tool like pytest. A typical pattern is to build a suite of tests for an app that ensure consistent functionality as the app evolves, and run the tests locally and/or in a CI environment like Github Actions.
+
+## The AppTest class
+
+<TileContainer>
+
+<RefCard href="/library/api-reference/app-testing/st.testing.v1.apptest" size="full">
+
+### st.testing.v1.AppTest
```

Comment:
Same feedback here as the one above about using `AppTest` rather than `st.testing.v1.AppTest` where we can